nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentManagerImpl.java

```diff
@@ -344,39 +301,42 @@ public DocumentModel createComment(DocumentModel docModel, DocumentModel parent,
-        Resource commentResource = relationManager.getResource(config.commentNamespace, comment, ctxMap);
-        if (commentResource == null) {
-            throw new NuxeoException("Could not adapt document model to relation resource ; "
-                    + "check the service relation adapters configuration");
-        }
-        Resource predicate = new ResourceImpl(config.predicateNamespace);
-
-        List<Statement> statementList = graph.getStatements(commentResource, predicate, null);
-        if (graph instanceof JenaGraph) {
-            // XXX AT: BBB for when repository name was not included in the
-            // resource uri
-            Resource oldDocResource = new QNameResourceImpl(config.commentNamespace, comment.getId());
-            statementList.addAll(graph.getStatements(oldDocResource, predicate, null));
-        }
-
-        List<DocumentModel> docList = new ArrayList<DocumentModel>();
-        for (Statement stmt : statementList) {
-            QNameResourceImpl subject = (QNameResourceImpl) stmt.getObject();
-            DocumentModel docModel = (DocumentModel) relationManager.getResourceRepresentation(config.documentNamespace,
-                    subject, ctxMap);
-            if (docModel == null) {
-                log.warn("Could not adapt comment relation subject to a document "
-                        + "model; check the service relation adapters configuration");
-                continue;
+        return CoreInstance.doPrivileged(comment.getRepositoryName(), session -> {
+            Map<String, Object> ctxMap = Collections.<String, Object> singletonMap(
+                    ResourceAdapter.CORE_SESSION_CONTEXT_KEY, session);
+            RelationManager relationManager = Framework.getService(RelationManager.class);
+            Graph graph = relationManager.getGraph(config.graphName, session);
+            Resource commentResource = relationManager.getResource(config.commentNamespace, comment, ctxMap);
+            if (commentResource == null) {
+                throw new NuxeoException("Could not adapt document model to relation resource ; "
+                        + "check the service relation adapters configuration");
+            }
+            Resource predicate = new ResourceImpl(config.predicateNamespace);
+
+            List<Statement> statementList = graph.getStatements(commentResource, predicate, null);
+            if (graph instanceof JenaGraph) {
+                // XXX AT: BBB for when repository name was not included in the
+                // resource uri
+                Resource oldDocResource = new QNameResourceImpl(config.commentNamespace, comment.getId());
+                statementList.addAll(graph.getStatements(oldDocResource, predicate, null));
             }
-            docList.add(docModel);
-        }
-        return docList;
 
+            List<DocumentModel> docList = new ArrayList<DocumentModel>();
```

Comment:
same as above, i know that is the old code, but i think is better to make it :)

Reply:
Oh didn't read that. Ok then I'll clean it all